manta-pay/src/crypto/ecc/arkworks.rs

```diff
@@ -601,4 +649,20 @@ mod test {
         println!("variable base mul constraint: {:?}", ctr2 - ctr1);
         println!("fixed base mul constraint: {:?}", ctr3 - ctr2);
     }
+
+    /// Checks if the windowed multiplcation is correct.
+    #[test]
+    fn windowed_mul_is_correct() {
+        let compiler = &mut ();
+        let scalar = Scalar::<Bls12_381_Edwards>::gen(&mut OsRng);
+        let point = Group::<Bls12_381_Edwards>::sample((), &mut OsRng);
+        const WINDOW_SIZE: usize = 4;
+        window_multiplication_correctness(
+            WINDOW_SIZE,
+            &scalar,
+            point,
+            |scalar, _| scalar.0.into_repr().to_bits_be(),
+            compiler,
+        );
```

Comment:
Also, can you add the circuit correctness check and the measurement check:
```suggestion
        window_multiplication_correctness(
            4,
            &Scalar::<Bls12_381_Edwards>::gen(&mut OsRng),
            Group::<Bls12_381_Edwards>::gen(&mut OsRng),
            |scalar, _| scalar.0.into_repr().to_bits_be(),
            &mut (),
        );
```

Reply:
Done, circuit correctness and measurement checks to come.